sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -116,8 +118,16 @@ def main(config:str,
 
     for bookpath in bookcart:
         if check_meta_type(bookpath) in accept_type:
-            #obsfiledb creation
-            checkbook(bookpath, config, add=True, overwrite=True)
+
+            try:
+                #obsfiledb creation
+                checkbook(bookpath, config, add=True, overwrite=True)
+            except Exception as e:
+                if config_dict["skip_bad_books"]:
+                    print(f"failed to add {bookpath}")
+                    continue
+                else:
+                    raise e
```

Comment:
Is there a way we can mark failed books so we don't continuously retry them?

Reply:
here, no. But I'm actively working on adding running of check_book to the end of the book binding process so hopefully they fail then when we're actually tracking fails